kratos/includes/constitutive_law.h

```diff
@@ -82,6 +82,11 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
     typedef std::size_t SizeType;
     typedef Geometry<Node < 3 > > GeometryType;
 
+    typedef array_1d<double, 6> StrainVectorType;
```

Comment:
Fo the moment this is OK, the pain will be actually replace the laws

Reply:
Are we going to consider always 3D by default?

Reply:
> Are we going to consider always 3D by default?

why not?
I mean this should have a size that can be used for everything

can you explain more your question?

Reply:
I agree you @philbucher , the obj is to use the max size andthen resize if it is lower dimension

Reply:
That was the question

Reply:
> 
> 
> > Another concern is what to do with the `CalculateValues`... Now I am forced to do a` static_cast<Vector>(rValues.GetStressVector)` in order to be consistent
> 
> I would update those methods as well, with the pain that involves...

Agree.

Reply:
If so, the `CAUCHY_STRESS_VECTOR` variable should be a `StressVectorType `and not a `Vector `anymore right?

Reply:
Indeed. Maybe we should discuss this in more detail